aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AmazonPinpointAnalyticsPluginConfiguration.java

```diff
@@ -18,32 +18,29 @@
 /**
  * Configuration options for Amplify Analytics Pinpoint plugin.
  */
-public final class AmazonPinpointAnalyticsPluginConfiguration {
+final class AmazonPinpointAnalyticsPluginConfiguration {
```

Comment:
Without specifying public, doesn't it default to package-private which leads consumers of Amplify not to be able to view this?

Reply:
Yes; this is the key question. Should a user have access to this object, or it better internal? Unless we have a use case in mind where the user should interact with this thing, lesser access is the better default.

Reply:
If we're making this change, so that the customer can only provide JSON configuration,  having a configuration object at all might be extraneous. We already don't have one in Storage and the one in API seems like an unnecessary extra step where we read the JSON into the configuration object only to then read the same data back out of the configuration object.

Reply:
In our design the Plugin implementations are a development entry point. Kind of like an Activity class in Android. So, you could easily build a functioning system by writing all of your code in this class. But, the implementation will only grows, as features / bugfixes come in. So, the best thing you can do is to defensively split stuff out of the plugin implementation from the get-go. A reasonable question to ask at each iteration is "how can I decouple my business logic from this plugin class?"